http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/BaseCloudServiceRuntimeMonitor.scala

```diff
@@ -479,7 +481,11 @@ abstract class BaseCloudServiceRuntimeMonitor[F[_]] {
           failedRuntime(
             monitorContext,
             runtimeAndRuntimeConfig,
-            RuntimeErrorDetails(s"${toolsStillNotAvailable} didn't start up properly", None, Some("tool_start_up")),
+            RuntimeErrorDetails(
+              s"${toolsStillNotAvailable.map(_.entryName).mkString(", ")} failed to start after ${monitorConfig.checkTools.interruptAfter.toMinutes} minutes.",
```

Comment:
Tried to improve this error message. It looks like this now:
![image](https://user-images.githubusercontent.com/5368863/135921539-1c1bd8df-fd0b-4c07-b5e6-e405a8b31a0d.png)

cc @alessanrdo

Reply:
so this is 10 min starts from when the VM becomes `Running` in GCP? do we need a bit more time just case if user's startup script takes a bit longer to run?

Reply:
I think if there is a user script, it waits for the user script to be done before moving on to `handleCheckTools`. See: https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/GceRuntimeMonitor.scala#L254. So user script execution would fall into the 20min bucket for VM creation.

Reply:
ah right...forgot about that...so this means if user's startup script is stuck, user will still need to wait....think this is still improvement, but just trying to understand the impact

Reply:
oh nvm, I think `checkAgain` has a timeout check as well...so this compliments that then

Reply:
Yeah it's tricky. I think we have the following situation:

* If there is a startup script:
   - 20m timeout for GCP VM creation, docker image download, startup script
   - 10m timeout for Jupyter startup

* If there is no startup script:
   - 20m timeout for GCP VM creation
   - 10m timeout for docker image download, Jupyter startup

For the second case it might be good to put `docker image download` in the 20m bucket, and try to use the 10m bucket _only_ for Jupyter startup. wdyt?

Reply:
it makes sense.. but when there's no startup script, how do you make sure `docker image download` happens before we check if tools are up?

Reply:
Yeah I'm not sure.. we'd need to design how that works. Maybe the init script could write a marker file to the staging bucket when `docker image download` is done.

Anyway maybe don't need to do that for this PR, but could ticket separately in case there are timeouts with custom images.

Reply:
I wonder if it makes sense to adjust `10m` bucket timeout based on whether there's startup script...if without, we make it 20m or sth?

Reply:
Hmm that could work. Basically we try and allocate `10m` for "customization" (e.g. docker download and user script), and put that 10m in either bucket depending on whether there's a user script present. Will see how this looks..

Reply:
I feel like this idea makes the logic a bit more complicated -- e.g. we're trying to infer where docker image pull happens depending on other fields. For Dataproc it's part of `Creating` GCP status; for GCE it happens after VM is `Running` but before the user script which may or may not be present, etc.

Leaning toward leaving this PR as is and adding explicit tracking of image pull if we need to (in a separate PR).

Reply:
sounds good...it might make sense to do what you mentioned in the previous comment about writing a marker file too